toolkit/tools/precacher/precacher.go

```diff
@@ -199,6 +199,11 @@ func monitorProgress(total int, results chan downloadResult, doneChannel chan st
 			lastProgressUpdate = progressPercent
 		}
 	}
+
+	if failed > 0 {
+		err = fmt.Errorf("failed to download %d package(s) out of %d total", failed, total)
```

Comment:
We might need to revisit this and make it configurable (something like `PRECACHER_ERROR="fatal"` vs `PRECACHER_ERROR="warn"`, defaulting to `"warn"`. That said, I can't think of any cases where the tool would fail here, and the fetcher tool would succeed (ignoring transient repo errors that might resolve between the two tools running).